speechbrain/dataio/dataio.py

```diff
@@ -28,6 +31,50 @@
 logger = logging.getLogger(__name__)
 
 
+def load_audio_pyav(uri, num_frames=-1, frame_offset=0, normalize=True):
```

Comment:
Maybe we could have a more generic `load_audio` function in SB that is not necessarily tied to PyAV and could ultimately be made to support different backends (e.g. still allowing to use torchaudio if desired)?

In theory PyAV should already sort of cover our usecases but it would still be more future-proof, I think.

Reply:
I do like this. In lhoste they have a function that switch the backend if one fails:
```
@lru_cache(maxsize=1)
def get_default_audio_backend() -> "AudioBackend":
    """
    Return a backend that can be used to read all audio formats supported by Lhotse.

    It first looks for special cases that need very specific handling
    (such as: opus, sphere/shorten, in-memory buffers)
    and tries to match them against relevant audio backends.

    Then, it tries to use several audio loading libraries (torchaudio, soundfile, audioread).
    In case the first fails, it tries the next one, and so on.
    """
    return CompositeAudioBackend(
        [
            # First handle special cases: OPUS and SPHERE (SPHERE may be encoded with shorten,
            #   which can only be decoded by binaries "shorten" and "sph2pipe").
            FfmpegSubprocessOpusBackend(),
            Sph2pipeSubprocessBackend(),
            # New FFMPEG backend available only in torchaudio 2.0.x+
            TorchaudioFFMPEGBackend(),
            # Prefer libsndfile for in-memory buffers only
            LibsndfileBackend(),
            # Torchaudio should be able to deal with most audio types...
            TorchaudioDefaultBackend(),
            # ... if not, try audioread...
            AudioreadBackend(),
            # ... oops.
        ]
    )
```

see: https://github.com/lhotse-speech/lhotse/blob/master/lhotse/audio/backend.py
 
We could have something more generic like that.

Reply:
I am against this option. More backend means more maintenance. We already had to manage one until now, it's already causing a mess.

Reply:
I think I'm with Titouan on this. Perhaps its worth noting that users are free to use lhotse with speechbrain if they want the more flexible audio loading.